client/state/comments/test/reducer.js

```diff
@@ -50,130 +42,85 @@ const commentsNestedTree = [
 describe( 'reducer', () => {
 	describe( '#items()', () => {
 		it( 'should build an ordered by date list', () => {
-			const res = items( undefined, {
+			const response = items( undefined, {
 				type: COMMENTS_RECEIVE,
 				siteId: 1,
 				postId: 1,
 				comments: [ ...commentsNestedTree ].sort( () => Math.random() * 2 % 2 ? -1 : 1 )
 			} );
+			const ids = map( response[ '1-1' ], 'ID' );
 
-			const specificRes = getPostCommentItems( { comments: { items: res } }, 1, 1 );
-
-			expect( specificRes.size ).to.equal( commentsNestedTree.length );
-
-			[ ...commentsNestedTree ].sort( ( a, b ) => new Date( b.date ) - new Date( a.date ) )
-							.forEach( ( comment, idx ) => expect( comment.ID ).to.equal( specificRes.getIn( [ idx, 'ID' ] ) ) );
+			expect( response[ '1-1' ] ).to.have.lengthOf( 6 );
+			expect( ids ).to.eql( [ 6, 7, 8, 9, 10, 11 ] );
 		} );
 
 		it( 'should build correct items list on consecutive calls', () => {
-			const res = items( undefined, {
-				type: COMMENTS_RECEIVE,
-				siteId: 1,
-				postId: 1,
-				comments: commentsNestedTree.slice( 0, 2 )
-			} );
+			const state = {
+				'1-1': commentsNestedTree.slice( 0, 2 )
+			};
 
-			const res2 = items( res, {
+			const response = items( state, {
 				type: COMMENTS_RECEIVE,
 				siteId: 1,
 				postId: 1,
 				comments: commentsNestedTree.slice( 1, commentsNestedTree.length )
 			} );
 
-			const specificCommentItemList = getPostCommentItems( { comments: { items: res2 } }, 1, 1 );
-
-			expect( specificCommentItemList.size ).to.equal( commentsNestedTree.length );
-			commentsNestedTree.forEach( ( comment, idx ) =>
-				expect( specificCommentItemList.getIn( [ idx, 'ID' ] ) ).to.equal( comment.ID )
-			);
+			expect( response[ '1-1' ] ).to.have.lengthOf( 6 );
 		} );
 
 		it( 'should remove a comment by id', () => {
 			const removedCommentId = 9;
-
-			const state = items( undefined, {
-				type: COMMENTS_RECEIVE,
-				siteId: 1,
-				postId: 1,
-				comments: commentsNestedTree
-			} );
-
-			const testedState = items( state, {
+			const state = { '1-1': commentsNestedTree };
```

Comment:
Do you mind `deepFreezing` state in these cases so we can detect any unwanted mutations?